rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.tcltestrun;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.maven.cli.MavenCli;
+import org.eclipse.m2e.core.internal.Bundles;
+import org.eclipse.m2e.core.internal.MavenPluginActivator;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Executes a maven build in a new jvm using the embedded maven. The maven
+ * embedder api allows a maven build without a m2 variable configuration.
+ *
+ */
+public class MavenExecutor {
+
+	private static Logger logger = LoggerFactory.getLogger(MavenExecutor.class);
+
+	/**
+	 * Executes the maven build using maven embedder. It allways starts with a
+	 * clean operation to delete old test results.
+	 * 
+	 * @param goal
+	 *            of maven to be exeucted.
+	 * @param pathToPom
+	 *            path to the folder where the pom.xml is located.
+	 * @return int with exit code
+	 * 
+	 */
+	public int execute(String goal, String pathToPom) {
+		System.setProperty("maven.multiModuleProjectDirectory", pathToPom);
+		MavenCli cli = new MavenCli();
+		int result = cli.doMain(new String[] { "clean", goal }, pathToPom, System.out, System.err);
```

Comment:
We don't need to change this now but we should keep in mind that we don't always perform a clean build.